### PR TITLE
Feature/fix hole visualizer triangle fan

### PR DESCRIPTION
**Pull request by lukaswagner**
can be merged now

### CODE REVIEW

File: src/plugins/nodeVisualizer/visualization/HoleVisualization.coffee [hunk 1]
```diff
@@ -1,5 +1,6 @@
 THREE = require 'three'
 threeHelper = require 'convertify/threeHelper'
+Util = require '../../platenerPipeline/Util'
```
Comment (mischkew) [inline src/plugins/nodeVisualizer/visualization/HoleVisualization.coffee]: dont require modules across plugins.
consider plugins as 3rd party packages, where they not necessarily ship together.
this means:
a) either build a separate utils npm package and add it to the overall dependencies
or b) put common utils into the convertify core (meaning place the file in the converitfy subdir)

File: src/plugins/nodeVisualizer/visualization/HoleVisualization.coffee [hunk 2]
```diff
@@ -27,58 +28,46 @@ class HoleVisualization
...
-    #     if edgeLoop.hole
-    #       material.polygonOffset = true
-    #       material.polygonOffsetFactor = -1
-    #       material.polygonOffsetUnits = 0.00001
-    #
-    #     mesh = new THREE.Mesh geomToDraw, material
-    #     drawable.add(mesh)
+    lines = new THREE.Object3D()
+    lineMaterial = new THREE.LineBasicMaterial(
+      { color: 0xffffff, depthTest: false })
```
Comment (mischkew) [inline src/plugins/nodeVisualizer/visualization/HoleVisualization.coffee]: no braces, no commas for objects -> new line each property
place closing parenthesis ")" on new line

File: src/plugins/nodeVisualizer/visualization/HoleVisualization.coffee [hunk 3]
```diff
@@ -27,58 +28,46 @@ class HoleVisualization
...
+        geomToDraw.applyMatrix(
+          new THREE.Matrix4()
+          .makeTranslation(0, 0, edgeLoop.xyPlaneVertices[0].z))
+        geomToDraw.applyMatrix(
+          Util.copyToMatrix4(shape.rotationMatrix.clone().transpose()))
 
         material = new THREE.MeshBasicMaterial(
-          color: if edgeLoop.hole then 0xff0000 else 0x00ff00,
-          side: 2 )
+          color: 0x00ff00, side: 2 )
```
Comment (mischkew) [inline src/plugins/nodeVisualizer/visualization/HoleVisualization.coffee]: no comma
")" on new line

**Comment by mischkew** [comment]
code understanding is hard when lots of nested structure (for, if, ...) are used next to each other.
consider inserting some linebreaks and explanatory comments.

File: src/plugins/nodeVisualizer/visualization/HoleVisualization.coffee [hunk 4]
```diff
@@ -27,58 +28,46 @@ class HoleVisualization
...
-        for vertex, vertexInd in edgeLoop.vertices
-          geomToDraw.vertices.push( vertex )
-
-          if vertexInd >= 2
-            face = new THREE.Face3( 0, vertexInd - 1, vertexInd )
-            geomToDraw.faces.push( face )
+        # start at last, iterate to each vertex
+        len = edgeLoop.xyPlaneVertices.length
+        shapeToDraw.moveTo edgeLoop.xyPlaneVertices[len - 1].x,
+          edgeLoop.xyPlaneVertices[len - 1].y
```
Comment (mischkew) [inline src/plugins/nodeVisualizer/visualization/HoleVisualization.coffee]: shapeToDraw.moveTo(
  edgeLoop...,
  edgeLoop...
)

File: src/plugins/nodeVisualizer/visualization/HoleVisualization.coffee [hunk 5]
```diff
@@ -27,58 +28,46 @@ class HoleVisualization
...
+        for vertex in edgeLoop.xyPlaneVertices
+          shapeToDraw.lineTo vertex.x, vertex.y
+        geomToDraw = new THREE.ShapeGeometry [shapeToDraw]
+        geomToDraw.applyMatrix(
+          new THREE.Matrix4()
+          .makeTranslation(0, 0, edgeLoop.xyPlaneVertices[0].z))
```
Comment (mischkew) [inline src/plugins/nodeVisualizer/visualization/HoleVisualization.coffee]: any line break with cascading function calls requires a "tab"
be consistent. omit parentheses for one-liners

```coffee
new THREE.Matrix4()
  .makeTranslation 0, 0, edgeLoop.xyPlaneVertices[0].z
```

**Comment by lukaswagner** [comment]
thanks for the feedback, i'll clean up the code tomorrow :)

**Comment by mischkew** [comment]
:+1:

**Pull request merged**